### PR TITLE
debugger: Implement F1 helper

### LINKED ISSUE #9101: [Feature Request] Document keyboard shortcuts on the debugger
The debugger has some features which most people who use dont know exist such as registers editor, instructions editor, SPU and RSX captures. because they cant know it, it only exists in code without any documentation.

### PR DESCRIPTION
**Pull request by elad335**
image

Fixes #9101

### CODE REVIEW

File: rpcs3/rpcs3qt/debugger_frame.cpp [hunk 1]
```diff
@@ -227,6 +227,38 @@ void debugger_frame::keyPressEvent(QKeyEvent* event)
 	const auto cpu = this->cpu.lock();
 	int i = m_debugger_list->currentRow();
 
+	switch (event->key())
+	{
+	case Qt::Key_F1:
+	{
+		QDialog* diag = new QDialog(this);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/debugger_frame.cpp:234]: why do you always call it diag..... this makes zero sense

File: rpcs3/rpcs3qt/debugger_frame.cpp [hunk 2]
```diff
@@ -227,6 +227,38 @@ void debugger_frame::keyPressEvent(QKeyEvent* event)
...
+		layout->addWidget(l);
+		diag->setLayout(layout);
+		diag->setFixedSize(diag->sizeHint());
+		diag->move(QCursor::pos());
+		diag->exec();
+		return;
+	}
+	}
+
 	if (!isActiveWindow() || !cpu)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/debugger_frame.cpp:262]: there's a reason why we return if this is not the active window...

Comment (elad335) [reply rpcs3/rpcs3qt/debugger_frame.cpp]: Fixed, I fixed it in other places as well.

File: rpcs3/rpcs3qt/debugger_frame.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/rpcs3qt/debugger_frame.cpp]: Fixed, I fixed it in other places as well.

File: rpcs3/rpcs3qt/debugger_frame.cpp [hunk 3]
```diff
@@ -227,7 +227,44 @@ void debugger_frame::keyPressEvent(QKeyEvent* event)
 	const auto cpu = this->cpu.lock();
 	int i = m_debugger_list->currentRow();
 
-	if (!isActiveWindow() || !cpu)
+	if (!isActiveWindow())
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/debugger_frame.cpp:230]: This should be at the beginning of the function, and the cpu and currentRow stuff after your new switch

Comment (elad335) [reply rpcs3/rpcs3qt/debugger_frame.cpp:227]: Done.

**Pull request merged**